pkg/internal/token/clientcertcredential.go

```diff
@@ -0,0 +1,206 @@
+package token
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity/cache"
+	"golang.org/x/crypto/pkcs12"
+)
+
+type ClientCertificateCredential struct {
+	cred *azidentity.ClientCertificateCredential
+}
+
+var _ CredentialProvider = (*ClientCertificateCredential)(nil)
+
+func newClientCertificateCredential(opts *Options) (CredentialProvider, error) {
+	if opts.ClientID == "" {
+		return nil, fmt.Errorf("client ID cannot be empty")
+	}
+	if opts.TenantID == "" {
+		return nil, fmt.Errorf("tenant ID cannot be empty")
+	}
+	if opts.ClientCert == "" {
+		return nil, fmt.Errorf("client certificate cannot be empty")
+	}
+	var (
+		c   azidentity.Cache
+		err error
+	)
+	if opts.UsePersistentCache {
+		c, err = cache.New(nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create cache: %s", err)
```

Comment:
nit: use %w to wrapp error

Reply:
no, we only have golangci-lint enabled at this moment. We should enable go vet etc

Reply:
i can enable it in next PR